plugins/postgres/dbt/adapters/postgres/connections.py

```diff
@@ -204,4 +204,7 @@ def get_response(cls, cursor) -> AdapterResponse:
 
     @classmethod
     def data_type_code_to_name(cls, type_code: int) -> str:
-        return string_types[type_code].name
+        if type_code in string_types:
+            return string_types[type_code].name
+        else:
+            return f"unknown type_code {type_code}"
```

Comment:
@rlh1994 [brought up](https://github.com/dbt-labs/dbt-core/pull/8887#issuecomment-1777754121) the idea of raising a debug-level warning in this case.

Seems like a valuable idea to explore. Maybe something like this?

```python
            logger.debug("unknown type_code: {}", type_code)
```

Reply:
@MichelleArk how would you feel about adding this ☝️ debug output when the `type_code` isn't found?

Reply:
It may be a little redundant given `unknown type_code` will appear in the error message either way. So I'd either omit it or write a more informative message like"type_code not found in adapter-provided lookup, defaulting to 'unknown type_code'"

And we may actually need to use fire_event to fire a debuglevel event as opposed not logger.debug

Reply:
👍 Splitting this into its own issue:
- https://github.com/dbt-labs/dbt-core/issues/8912 

That way we can ship this fix while still giving that debug message enough attention.